id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -530,22 +533,32 @@ private void updateRequestBodyData(IdRequestDTO request) throws IdRepoAppExcepti
 		}
 	}
 
+
 	@SuppressWarnings({ "rawtypes", "unchecked" })
-	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) throws IdRepoAppException {
+	protected void updateVerifiedAttributes(
+			IdRequestDTO<T> requestDTO,
+			DocumentContext inputData,
+			DocumentContext dbData,
+			boolean isV2Version,
+			Map<String, Object> identityMap
+	) throws IdRepoAppException {
 		List dbVerifiedAttributes = (List) dbData.read(DOT + VERIFIED_ATTRIBUTES);
-		dbVerifiedAttributes.remove(null);
-		boolean isV2Version = requestDTO.getVerifiedAttributes() instanceof Map ? true : false;
 		if (dbVerifiedAttributes.isEmpty()) {
-			dbVerifiedAttributes.add(isV2Version ? new HashMap<>() : new ArrayList<>());
+			dbVerifiedAttributes.add(isV2Version ? new HashMap<>() :  new ArrayList<>());
 		}
 		if (isV2Version) {
-			Map dbVerifiedAttributeMap = (Map) dbVerifiedAttributes.get(0);
-			Map<String, Object> identityMap = idRepoServiceHelper.convertToMap(requestDTO.getIdentity());
-			dbVerifiedAttributeMap.keySet().removeIf(identityMap::containsKey);
-			if (Objects.nonNull(requestDTO.getVerifiedAttributes())
-					&& !((Map) requestDTO.getVerifiedAttributes()).isEmpty()) {
-				dbVerifiedAttributeMap.putAll((Map) requestDTO.getVerifiedAttributes());
+			Map<String, Object> dbVerifiedAttributeMap = (Map<String, Object>) dbVerifiedAttributes.get(0);
```

Comment:
If the dbVerifiedAttributes is empty, you are setting it to an empty collection. Now, in this line, if it's a v2 endpoint, and if you try accessing the "0" index element, it will throw an AIOB exception.